Feature/re factoring with vuejs framework

punalsc opened the pull request:
@gtvj  , could you have a look at this PR.
Here is a link for you to view.

http://test.nationalarchives.gov.uk/prototype/tna-wifi-prototype/2/

punalsc committed:
babel and grunt integration

punalsc committed:
Added grunt and babel configurations.

punalsc committed:
added sass,cssmin and ugligy to grunt tasks

punalsc committed:
grunt minification

punalsc committed:
refactoring js to EM6

punalsc committed:
returning the methods

punalsc committed:
minor bug fixes

punalsc committed:
refactoring code

punalsc committed:
refactoring form submit with validation

punalsc committed:
css cleanup

punalsc committed:
updated git ignore

punalsc committed:
one submit button for two forms

punalsc committed:
added initialization

punalsc committed:
email validation

punalsc committed:
added more validations

punalsc committed:
whitespace validation

punalsc committed:
added more validations

punalsc committed:
refactored validation

punalsc committed:
editorial changes

punalsc committed:
js refactoring

punalsc committed:
ajax form

punalsc committed:
peer review update

punalsc committed:
css fixes for logo

punalsc committed:
new layout setup

punalsc committed:
new slash screen updates

punalsc committed:
appended form

punalsc committed:
toggle terms and conditions

punalsc committed:
minor changes

punalsc committed:
email validation function

punalsc committed:
added email validation

punalsc committed:
accessibility css fixes

punalsc committed:
fixed tabindex

punalsc committed:
re-factored project structure

punalsc committed:
fixed minor css issues

punalsc committed:
refactored sass

punalsc committed:
responsive fixes

punalsc committed:
changed editorial text

netdesignr reviewed (changes requested):
Hi @punalsc,

Please find below my comments:
1. The link to the peer review https://github.com/nationalarchives/development-guide/blob/master/2017/tna-public-wifi-login.md
2. Functionality review:
    2.1. Although the vue.js integration is looking good have you considering adding unit tests? The application is quite small so maybe this is the right time to get used to the unit tests around vue.js? Here's a link to their documentation https://vuejs.org/v2/guide/unit-testing.html
    2.2. I'm afraid there's lack of documentation around the app.js which is mainly the Vue framework. Please consider adding comments to all functions/methods/variables wherever it's needed with a clear description.
3. I can confirm that the application was tested using the Vue browser debugging tool as well as cross-browser compatibility, with and without the JS.

punalsc committed:
re-ordered grunt tasks.

punalsc commented:
@netdesignr ,

I have updated the peer review https://github.com/nationalarchives/development-guide/blob/master/2017/tna-public-wifi-login.md.

I did integrate unit tests, but it was causing some errors during the setup, but because of time constraints, i had to take it out, but do plan to look further into it at a later date.

I did have comments to explain what the add was doing, but unfortunately babel was transpiling the source file, which is now fixed.

punalsc committed:
added comments to app
